validator/must_present.go

```diff
@@ -0,0 +1,51 @@
+package validator
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/spf13/cobra"
+)
+
+type MustPresentRule struct {
+	Fields []string
+}
+
+func (p *MustPresentRule) Validate(cmd *cobra.Command) []error {
+
+	var errors []error
+	err := validateMustPresent(cmd, p)
+	errors = append(errors, err...)
+
+	for _, child := range cmd.Commands() {
+		err := validateMustPresent(child, p)
+		errors = append(errors, err...)
+	}
+
+	return errors
+}
+
+func validateMustPresent(cmd *cobra.Command, p *MustPresentRule) []error {
+	var errors []error
+
+	for _, field := range p.Fields {
+		reflectValue := reflect.ValueOf(cmd).Elem().FieldByName(field)
+		errors = append(errors, validateByType(&reflectValue, field, cmd.CommandPath())...)
+	}
+	return errors
+}
+
+func validateByType(reflectValue *reflect.Value, field string, path string) []error {
```

Comment:
What this does? Not sure why we need it

Reply:
cobra.Command struct has different types of attributes. Hence to address all them. I did this. 
Like something is string, bool, []string, slice, func 
eg - Args is func, SilenceUsage is bool

by writing this function we can handle `mustPresent` for all the types present in cobra.Command

Reply:
ok